### ISSUE TITLE
ArgumentOutOfRangeException: T2.WeaponWheel2.SelectSlot

### ISSUE THREAD

**Issue by ChrisNZL** [ChrisNZL/Tallowmere2#720]
Auto bug report. v0.2.0.

Feedback ID: 20201210-R2XJX

```
9:04:25, Frame 340600, EXCEPTION »  ArgumentOutOfRangeException: Index was out of range. Must be non-negative and less than the size of the collection.
Parameter name: index
>>>>> CRITICAL ERROR >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>

System.ThrowHelper.ThrowArgumentOutOfRangeException (System.ExceptionArgument argument, System.ExceptionResource resource)
System.ThrowHelper.ThrowArgumentOutOfRangeException ()
T2.WeaponWheel2.SelectSlot (T2.WeaponWheel2Slot newSlot)
T2.WeaponWheel2.ProcessWeaponWheelInput (T2.Player player)
T2.Player.Update ()

GameStates: RompingThroughDungeon
GameSetupMode: NetworkCoop
Players: 2
DungeonRoomTransitionerState: Null
Room: 11 / 11
RoomModifiers: Bloatastic (Bloatastic), HealthRegen (Healthy Monsters)
YouHaveDiedState: Null
SystemPlayer InputDevice: XInput Controller
HumanPlayer 1 InputDevice: XInput Controller
```

**Comment by ChrisNZL** [comment]
Additional auto report. 0.2.3. Feedback ID: 20210129-9BK2R

```
3:24:44, Frame 187081, EXCEPTION »  ArgumentOutOfRangeException: Index was out of range. Must be non-negative and less than the size of the collection.
Parameter name: index
>>>>> CRITICAL ERROR >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>

System.ThrowHelper.ThrowArgumentOutOfRangeException (System.ExceptionArgument argument, System.ExceptionResource resource)
System.ThrowHelper.ThrowArgumentOutOfRangeException ()
T2.WeaponWheel2.SelectSlot (T2.WeaponWheel2Slot newSlot)
T2.WeaponWheel2.ProcessWeaponWheelInput (T2.Player player)
T2.Player.Update ()

GameStates: InitialisingAsNetworkHost, RompingThroughDungeon
OnlineGameInfo: Device is online.
Server: HostUS-HongKong-01
GameID: 2MDPN
GameSetupMode: NetworkCoop
Players: 2
DungeonRoomTransitionerState: Null
Room: 21 / 21
RoomModifiers: PoisonNightmare (剧毒梦魇), ManyElites (超多精英怪), LovelyLoot (丰厚战利品), LargeRoom (大型房间)
YouHaveDiedState: Null
SystemPlayer InputDevice: 键盘
HumanPlayer 1 InputDevice: 键盘
```

**Comment by ChrisNZL** [comment]
Handled better in 0.2.4.

Method to select the slot / weapon now checks if the desired slot is not found in the list before accessing, or if the index is higher than the list.Count.

Possibly related issue: [Garbage collection #853](https://github.com/ChrisNZL/Tallowmere2/issues/853)

Also worth noting: this issue happens for both online and offline play. Maybe the Weapon Selector list is not getting updated properly at times. It's a fairly rare/uncommon bug though. Not 100% sure.

**Comment by ChrisNZL** [comment]
Reopening this.

Whilst diving into the upcoming Weapon Selector work (choosing between dynamic, radial, and grid layouts), believe I came across this issue where:

- Internal WeaponWheel / Weapon Selector's weapon cache can get outdated if an item is sold, I think? So I want to try and reinvestigate and reproduce.

- Auto report: 0.2.6a, feedback ID: 20210422-69WQG

```
9:16:36, Frame 217049, EXCEPTION »  ArgumentOutOfRangeException: Index was out of range. Must be non-negative and less than the size of the collection.
Parameter name: index
>>>>> CRITICAL ERROR >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>

System.ThrowHelper.ThrowArgumentOutOfRangeException (System.ExceptionArgument argument, System.ExceptionResource resource)
System.ThrowHelper.ThrowArgumentOutOfRangeException ()
T2.WeaponWheel2.SelectSlot (T2.WeaponWheel2Slot newSlot)
T2.WeaponWheel2.ProcessWeaponWheelInput (T2.Player player)
T2.Player.Update ()

GameStates: RompingThroughDungeon
OnlineGameInfo: Device is online.
Server: Vultr-Tokyo-01
GameID: Y9MQV
GameSetupMode: NetworkCoop
Players: 2
DungeonRoomTransitionerState: Null
Room: 36 / 36
RoomModifiers: Spiky (满屋尖刺), MoreGoldOnDeath (财富), FlailsOnly (链枷派对), NoWaypoint (没有传送点), ManyElites (超多精英怪)
YouHaveDiedState: Null
SystemPlayer InputDevice: XInput Controller
HumanPlayer 1 InputDevice: XInput Controller
```

**Comment by ChrisNZL** [comment]
Should be fixed in 0.2.7, especially with the new Weapon Selector options.

Added further checks before accessing the index.
